### PR TITLE
WIP Partial fix for #8735 Failed Invites

### LINKED ISSUE #8735: Failed party invitations do not indicate which invitations failed
### General Info
  * UUID: 30983c37-52c3-415d-977c-38af6c7db91c

### Description
Party, Invite Friends, Invite Existing Users. Invitations can be sent to multiple users via the '+' button. However, if any invitation should fail, the standard message *User already in a party* pops up. The standard message does not indicate which user(s) invitation failed (which user is already in a party) nor does it indicate how many invitations failed (I tested this by sending invitations to two users who are already in other parties, and only one failure message popped up).

_edit by Alys 2018-02-20:_

PR #9939 has done a partial fix for this and we now see a single error message if one invitation has failed. However if two or more have failed, only the first error is reported. See the comments in this issue for details.  

Some of the comments mention using bluebird but we're moving away from it so best to not use it for this issue.

Some of the comments below will be related to the partial fix that was done in #9939 so they won't be relevant to the remaining work. Post here if you aren't sure about any comments!

### PR DESCRIPTION
**Pull request by JoshSears**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Partial Fix #8735

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
Altered /test/api/v3/integration/groups/POST-groups_invite.test.js to reflect new output
Altered /website/common/locales/en/groups.json - Added username and UUID outputs to failed invites messages
Altered /website/server/controllers/api-v3/groups.js - sends username and uuid to error messages

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 348b9f3a-83ef-4865-b4d1-8f64870c9734

### CODE REVIEW

File: website/server/controllers/api-v3/groups.js [hunk 1]
```diff
@@ -904,7 +893,7 @@ api.removeGroupMember = {
       throw new NotFound(res.t('groupMemberNotFound'));
     }
 
-    let message = req.query.message || req.body.message;
+    let message = req.query.message;
```
Comment (TheHollidayInn) [inline website/server/controllers/api-v3/groups.js]: Why did you remove the body? We do need that.

File: website/server/controllers/api-v3/groups.js [hunk 2]
```diff
@@ -861,7 +850,7 @@ api.removeGroupMember = {
       group.memberCount -= 1;
       if (group.hasNotCancelled())  {
         await group.updateGroupPlan(true);
-        await payments.cancelGroupSubscriptionForUser(member, group, true);
+        await payments.cancelGroupSubscriptionForUser(member, group);
```
Comment (TheHollidayInn) [inline website/server/controllers/api-v3/groups.js]: Why did you remove the true argument?

File: website/server/controllers/api-v3/groups.js [hunk 3]
```diff
@@ -759,19 +755,12 @@ api.leaveGroup = {
...
-      user.guilds.splice(guildIndex, 1);
-    }
+    if (group.hasNotCancelled())  await group.updateGroupPlan(true);
 
-    let isMemberOfGroupPlan = await user.isMemberOfGroupPlan();
-    if (!isMemberOfGroupPlan) {
-      await payments.cancelGroupSubscriptionForUser(user, group);
-    }
+    _removeMessagesFromMember(user, group._id);
+
```
Comment (TheHollidayInn) [inline website/server/controllers/api-v3/groups.js]: Can you add the membercheck. It is out of scope of this PR. It is possible the logic is duplicate, but the check is different then below, so we should have test coverage and check for duplication separately.

File: website/server/controllers/api-v3/groups.js [hunk 4]
```diff
@@ -336,11 +336,7 @@ api.getGroups = {
     }
 
     if (req.query.search) {
-      filters.$or = [];
-      const searchWords = req.query.search.split(' ').join('|');
-      const searchQuery = { $regex: new RegExp(`${searchWords}`, 'i') };
-      filters.$or.push({name: searchQuery});
-      filters.$or.push({description: searchQuery});
+      filters.$text = { $search: req.query.search };
```
Comment (TheHollidayInn) [inline website/server/controllers/api-v3/groups.js]: Please add the $or queries back. The text index does not work for our requirements.

**Comment by TheHollidayInn** [comment]
Thanks for the PR! You have a few edits that are out of scope a bit. I left comments on those.

**Comment by JoshSears** [comment]
The edits you pointed out were not (purposely) done.  Let me try updating my repo to see if the version I had was not up to date.

**Comment by paglias** [comment]
Looks like there are some issues with the modified files @Malcorr

**Comment by JoshSears** [comment]
@paglias OK, just got the message.  Will update my branch and resubmit.

**Comment by paglias** [comment]
thanks!

**Comment by JoshSears** [comment]
@paglias There are A LOT of merge conflicts now.  Trying to figure out the best way to do this, short of making a new branch.

**Comment by paglias** [comment]
@Malcorr in this case creating a new branch might be the best way to proceed :)

**Comment by JoshSears** [comment]
@paglias When I submit the PR, will it be for the original issue number (8735) or the number associated with this (9744)?

**Comment by paglias** [comment]
It should be for the original issue but if you reference this pr in the
description it'll be useful

Il 03 Feb 2018 18:19, "Josh Sears" <notifications@github.com> ha scritto:

**Comment by JoshSears** [comment]
New PR created after creating new branch due to too many merge conflicts, #9939
